### PR TITLE
[WIP] cloud APis

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/dao/RestDAO.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// @flow
+
+import Subscription from './Subcription';
+
+class RestDAO extends Subscription {
+  deleteByID() {}
```
Comment (jlkalberer) [inline src/dao/RestDAO.js]: Are you going to inherit from this? Like `ProductFirmwareDAO extends RestDAO`?

Comment (AntonPuko) [reply src/dao/RestDAO.js]: yeah, I thought about that, what is the sense of RestDAO if not that? only keeping cache things?

File: src/dao/ProductDeviceDAO.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// @flow
+
+class ProductDeviceDAO {
+  // api/v2/products/{idOrSlug}/devices
+  fetchMany() {}
+
+  // api/v2/products/{idOrSlug}/devices/{particleID}
```
[the remark this answers is not in the document]
Comment (jlkalberer) [reply src/dao/ProductDeviceDAO.js]: `fetchMany` should look like
```
fetchMany<TQueryParameters>(path: string, queryParameters: TQueryParameters): LoadObject<Array<TEntity>>
```

File: src/dao/RestDAO.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        .then(this._updateCacheForEntity)
+        .catch(error => {
+          this._updateCacheForError(stringifiedId, error);
+        });
+    }
+
+    return nullthrows(this._entityLoaderById.get(stringifiedId));
+  }
+
+  __fetchOnce<TQueryParams: Object>(
```
Comment (AntonPuko) [inline src/dao/RestDAO.js]: so, i made this for things like ping() , where we need to return new result every time. does it makes sense? Probably not the best method name though.

Comment (AntonPuko) [reply src/dao/RestDAO.js]: @jlkalberer but is the method logic right? the only way we can refetch every time is like we do for udpate request -using and returning clientID, right?

Comment (AntonPuko) [reply src/dao/RestDAO.js]: since  if we do it without clientID it will override previous request, which result can be still used somwhere in the UI

**Pull request merged**